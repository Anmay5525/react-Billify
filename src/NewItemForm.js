const { useRef } = React;

export default function NewItemForm({ handleNewItem }) {
  const name = useRef();
  const amount = useRef();
  const description = useRef();

  return (
    <div className="items-form">
      <div className="input-cnt">
        <div>Name</div>
        <input
          ref={name}
          className="input"
          id="name"
          type="text"
          placeholder="Enter name of item"
        />
      </div>

      <div className="input-cnt">
        <div>Amount</div>
        <input
          ref={amount}
          className="input"
          type="number"
          id="amount"
          placeholder="Enter amount"
        />
      </div>

      <div className="input-cnt">
        <div>Description</div>
        <textarea
          ref={description}
          className="textarea"
          id="description"
          placeholder="Add description"
          rows="4"
        ></textarea>
      </div>
      <div>
        <button
          className="items-add-btn"
          onClick={() =>
            handleNewItem(
              name.current.value,
              amount.current.value,
              description.current.value
            )
          }
        >
          Add Item
        </button>
      </div>
    </div>
  );
}
