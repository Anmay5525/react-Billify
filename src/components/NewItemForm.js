import React from "react";
import { toast } from "react-toastify";
import { useHistory } from "react-router-dom";

export default function NewItemForm() {
  const history = useHistory();
  const handleNewItem = (e) => {
    e.preventDefault();

    const url = "https://rzp-training.herokuapp.com/team1/items";

    const data = {
      name: e.target.name.value,
      amount: e.target.amount.value * 100,
      description: e.target.description.value,
      currency: "INR",
    };

    fetch(url, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(data),
    })
      .then((res) => {
        if (res.ok) {
          return res.json();
        }
        // return toast.error(
        //   `Something went wrong. Server responded with status ${res.status}`
        // );
        return null;
      })
      .then((r) => {
        if (r) {
          toast.success("New item created");
          history.push("/Items");
          // handleSubRouteChange("list");
        } else {
          toast.error(`Something went Wrong !`);
        }
      })
      .catch((error) => console.log(error));
  };

  return (
    <div className="content">
      <form className="items-form" onSubmit={(e) => handleNewItem(e)}>
        <div className="input-cnt">
          <div>Name</div>
          <input
            className="input"
            name="name"
            type="text"
            placeholder="Enter name of item"
            required
            maxLength="40"
          />
        </div>

        <div className="input-cnt">
          <div>Amount</div>
          <input
            className="input"
            type="text"
            pattern="^[0-9]+"
            name="amount"
            placeholder="Enter amount"
            required
            onInput={(e) => {
              if (e.target.validity.patternMismatch) {
                e.target.setCustomValidity("Enter a valid amount!");
              } else {
                e.target.setCustomValidity("");
              }
            }}
          />
        </div>

        <div className="input-cnt">
          <div>Description</div>
          <textarea
            className="textarea"
            name="description"
            placeholder="Add description"
            rows="4"
          />
        </div>
        <div>
          <button type="submit" className="items-add-btn">
            Add Item
          </button>
        </div>
      </form>
    </div>
  );
}

// NewItemForm.propTypes = {
//   handleSubRouteChange: PropTypes.func.isRequired,
// };
