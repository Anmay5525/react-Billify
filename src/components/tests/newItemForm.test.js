import React from 'react';
import renderer from 'react-test-renderer';
import NewItemForm from '../NewItemForm';

it("newItemForm renders without crashing",()=>{
    const tree=renderer.create(<NewItemForm/>).toJSON();
    expect(tree).toMatchSnapshot();
});