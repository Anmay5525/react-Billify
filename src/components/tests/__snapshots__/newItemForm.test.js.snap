// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newItemForm renders without crashing 1`] = `
<div
  className="content"
>
  <h2
    className="new-item-title"
  >
    New Item
  </h2>
  <div
    className="items-form-container"
  >
    <form
      className="items-form"
      onSubmit={[Function]}
    >
      <div
        className="input-cnt"
      >
        <div>
          Name
        </div>
        <input
          className="input"
          maxLength="40"
          name="name"
          placeholder="Enter name of item"
          required={true}
          type="text"
        />
      </div>
      <div
        className="input-cnt"
      >
        <div>
          Amount
        </div>
        <input
          className="input"
          name="amount"
          onInput={[Function]}
          pattern="^[0-9]+"
          placeholder="Enter amount"
          required={true}
          type="text"
        />
      </div>
      <div
        className="input-cnt"
      >
        <div>
          Description
        </div>
        <textarea
          className="textarea"
          name="description"
          placeholder="Add description"
          rows="4"
        />
      </div>
      <div>
        <button
          className="items-add-btn"
          type="submit"
        >
          Add Item
        </button>
      </div>
    </form>
  </div>
</div>
`;
